numba/tests/test_parfors.py

```diff
@@ -3554,6 +3554,21 @@ def test_impl(x, y):
         self.prange_tester(test_impl, X, Y, scheduler_type='unsigned',
                            check_fastmath=True, check_fastmath_result=True)
 
+    def test_prange29(self):
+        # issue7630: SSA renaming in prange header
+        def test_impl(flag):
+            result = 0
+            if flag:
+                for i in range(1000000):
```

Comment:
```suggestion
                for i in range(1):
```
The reproducer doesn't actually need to execute more than one iteration for the bug in #7630 to manifest.